src/main/java/com/laytonsmith/core/AliasCore.java

```diff
@@ -215,127 +201,123 @@ public final void reload(MCPlayer player, String[] settings, boolean firstLoad)
 
-			// PacketJumper.startup(); we're not using this yet
-			if(options.reloadExtensions()) {
-				ProfilePoint extensionManagerStartup = parent.profiler.start("Extension manager startup", LogLevel.VERBOSE);
-				try {
-					ExtensionManager.Startup();
-				} finally {
-					extensionManagerStartup.stop();
-				}
-			}
-			MSLog.GetLogger().Log(MSLog.Tags.GENERAL, LogLevel.VERBOSE, "Scripts reloading...", Target.UNKNOWN);
-			if(parent.persistenceNetwork == null || options.reloadPersistenceConfig()) {
-				ProfilePoint persistenceConfigReload = parent.profiler.start("Reloading persistence configuration", LogLevel.VERBOSE);
-				try {
-					MemoryDataSource.ClearDatabases();
-					ConnectionMixinFactory.ConnectionMixinOptions mixinOptions = new ConnectionMixinFactory.ConnectionMixinOptions();
-					mixinOptions.setWorkingDirectory(MethodScriptFileLocations.getDefault().getConfigDirectory());
-					parent.persistenceNetwork = new PersistenceNetworkImpl(MethodScriptFileLocations.getDefault().getPersistenceConfig(),
-							new URI("sqlite:/" + MethodScriptFileLocations.getDefault().getDefaultPersistenceDBFile()
-									.getCanonicalFile().toURI().getRawSchemeSpecificPart().replace('\\', '/')), mixinOptions);
-				} finally {
-					persistenceConfigReload.stop();
-				}
-			}
+			// newEnv is preserved here for a reusable environment for aliases
 			try {
-				parent.profiles = new ProfilesImpl(MethodScriptFileLocations.getDefault().getProfilesFile());
-			} catch (IOException | Profiles.InvalidProfileException ex) {
-				MSLog.GetLogger().e(MSLog.Tags.GENERAL, ex.getMessage(), Target.UNKNOWN);
-				return;
-			}
-			GlobalEnv gEnv = new GlobalEnv(parent.executionQueue,
-					MethodScriptFileLocations.getDefault().getConfigDirectory(), EnumSet.of(RuntimeMode.EMBEDDED));
-			StaticRuntimeEnv staticRuntimeEnv = new StaticRuntimeEnv(parent.profiler,
-					parent.persistenceNetwork, parent.profiles, new TaskManagerImpl());
-			gEnv.SetLabel(Static.GLOBAL_PERMISSION);
-			if(options.reloadExecutionQueue()) {
-				ProfilePoint stoppingExecutionQueue = parent.profiler.start("Stopping execution queues", LogLevel.VERBOSE);
-				try {
-					parent.executionQueue.stopAllNow();
-				} finally {
-					stoppingExecutionQueue.stop();
-				}
-			}
-			CommandHelperEnvironment cEnv = new CommandHelperEnvironment();
-			Environment env = Environment.createEnvironment(gEnv, staticRuntimeEnv, cEnv, compilerEnv);
-			if(options.reloadGlobals()) {
-				ProfilePoint clearingGlobals = parent.profiler.start("Clearing globals", LogLevel.VERBOSE);
-				try {
-					Globals.clear();
-				} finally {
-					clearingGlobals.stop();
-				}
-			}
-			if(options.reloadTimeouts()) {
-				ProfilePoint clearingTimeouts = parent.profiler.start("Clearing timeouts/intervals", LogLevel.VERBOSE);
-				try {
-					Scheduling.ClearScheduledRunners();
-				} finally {
-					clearingTimeouts.stop();
-				}
+				env = newEnv.clone();
```

Comment:
I'm not understanding why `newEnv` is cloned here, right after it has been created. Why is `newEnv` itself used as the environment, but `env` is used for the whole compilation and setup process of the MS code?

Reply:
This stores the new environment (if compilation is successful) in AliasCore for usage in alias() and elsewhere, though StaticRuntimeEnv is the important bit.  Before it would create a new StaticRuntimeEnv from fields stored in CommandHelperPlugin. So, I'm cloning here to better ensure the compilation and execution of the ms files doesn't modify the environment. Maybe we only need to store StaticRuntimeEnv in AliasCore, though. But there are some things that arguably should be moved from GlobalEnv to StaticRuntimeEnv but haven't yet, like the previously mentioned execution queue and the uncaughtExceptionHandler.

Reply:
I'm thinking moving the execution queue and uncaught exception handler to StaticRuntimeEnv is the appropriate move here. Are there others that need to be moved too? Then we can just generate a new GlobalEnv for aliases and avoid cloning.

Reply:
I believe that the uncaught exception handler currently works per compilation unit (alias, main ms file). This might have been changed in the past though, but it would be good to check that. Moving it to StaticRuntimeEnv would share it across all compilation units. I agree that the execution queue would fit the StaticRuntimeEnv.

Regarding the cloned environment: I think that if the compiler makes changes to an environment that is not going to be used, that we are looking at a bug there. Especially if this is just about the uncaught exception handler and the execution queue, which both cannot be set/changed/used in compile time as far as I am aware.

Reply:
May be useful to refer to #1156. It even links a PR where you've actually tried to fix this issue before.

Reply:
Oh wow, that was 7 years ago, I doubt that I knew what I was doing exactly. But we both had the idea that the uncaught exception handler should be shared, so maybe this is the time to finally do that by doing what you proposed. There was no StaticRuntimeEnv back then, so the solution was not obvious before. Maybe @LadyCailin wants to share an opinion on making the uncaught exception handler shared among all compilation units as well?

Reply:
Yes, moving this into the StaticRuntimeEnv makes sense, now that that is a thing.